src/ScottPlot/plottables/PlottableScatterHighlight.cs

```diff
@@ -0,0 +1,118 @@
+using ScottPlot.Drawing;
+using System.Diagnostics;
+using ScottPlot.plottables;
+
+namespace ScottPlot
+{
+    public class PlottableScatterHighlight : PlottableScatter, IExportable, IHighlightable
+    {
+        protected List<int> highlightedIndexes;
+        public MarkerShape highlightedShape;
+        public float highlightedMarkerSize;
+        public Color highlightedColor;
+        private Pen penLineError;
+
+        public PlottableScatterHighlight(double[] xs, double[] ys, Color color, double lineWidth, double markerSize, string label,
+            double[] errorX, double[] errorY, double errorLineWidth, double errorCapSize, bool stepDisplay, MarkerShape markerShape, LineStyle lineStyle, MarkerShape highlightedShape, Color highlightedColor, double highlightedMarkerSize)
+            : base(xs, ys, color, lineWidth, markerSize, label, errorX,  errorY, errorLineWidth, errorCapSize, stepDisplay, markerShape, lineStyle)
+        {
+            this.highlightedColor = highlightedColor;
+            this.highlightedMarkerSize = (float)highlightedMarkerSize;
+            this.highlightedShape = highlightedShape;
+            this.highlightedIndexes = new List<int>();
+        }
+
+        protected override void DrawPoint(Settings settings, List<PointF> points, int i)
+        {
+            if (highlightedIndexes.BinarySearch(i) >= 0) //Always returns negative number for item not found: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.binarysearch?view=netcore-3.1#System_Collections_Generic_List_1_BinarySearch__0_
+            {
+                MarkerTools.DrawMarker(settings.gfxData, points[i], highlightedShape, highlightedMarkerSize, highlightedColor);
+            }
+            else
+            {
+                base.DrawPoint(settings, points, i);
+            }
+        }
+
+        public void HighlightClear()
+        {
+            highlightedIndexes.Clear();
+        }
+
+        public void HighlightPoint(int index)
+        {
+            int insertPosition = highlightedIndexes.BinarySearch(index);
+            if (insertPosition < 0) {
+                insertPosition = ~insertPosition;   //This might look like witchcraft
+                                                    //However, List<T>.BinarySearch returns the ones-compliment (bitwise inverse) of the index of the next-higher value in the list
+                                                    //if the value is not part of the list.
+                                                    //If there is no higher value it returns the ones-compliment of this.Count()
+                                                    //Source: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.binarysearch?view=netcore-3.1#System_Collections_Generic_List_1_BinarySearch__0_
+            }
+            highlightedIndexes.Insert(insertPosition, index);
+        }
+
+        private int GetPointNearestIndex(double x) {
```

Comment:
I think this method name is trying to be `GetIndexNearestPoint()`, or `GetIndexNearestX()`. The second option would fix the overload issue that would arise if you create a `GetIndexNearestY()` too (do you think we should have one?)

Reply:
I don't know how useful `GetIndexNearestY()` is, but I added it. This is mostly for following the mouse cursor around, and I think most users would prefer to snap to either the closest point, or the closest point to the x span of the cursor. However, it's entirely possible that someone could find a use for it outside of mouse controls

Reply:
Sounds good! A use case might be finding the nearest bar graph to the cursor when horizontal bars are used if later we make bar graphs `IHighlightable`

https://swharden.com/scottplot/cookbooks/4.0.32/#plottypes-bar-horizontal-bar-graph

Now seems like the best time to go ahead and add X and Y methods to `IHighlightable`